gammapy/time/lightcurve.py

```diff
@@ -314,11 +321,204 @@ def create_fixed_time_bin(time_step, spectrum_extraction):
+            Contains a list of time (in MJD) to break the current point creation
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of time intervals  and information about their content : on/off events, alpha, significance
+
+        Examples
+        --------
+        extract intervals for light curve :
+            intervals = list(zip(table['t_start'], table['t_stop']))
+
+        """
+
+        time_holder = []
+        obs_properties = []
+        n_obs = 0
+
+        if separators is not None:
+            for time in separators:
+                time_holder.append([Time(time, format='mjd').tt.mjd, 'break'])
+
+        for obs in spectrum_extraction.obs_list:
+            # start and end will need to be redefined once the data storage format is fixed
+            time_holder.append([obs.events.observation_time_start.tt.mjd, 'start'])
+            time_holder.append([obs.events.observation_time_end.tt.mjd, 'end'])
+            obs_properties.append(dict(deadtime=obs.observation_dead_time_fraction,
+                                       A_off=spectrum_extraction.bkg_estimate[n_obs].a_off))
+            n_obs += 1
+        obs_properties = Table(rows=obs_properties)
+
+        # prepare the on and off photon list as in the flux point computation -> should be updated accordingly
+        for t_index, obs in enumerate(self.obs_list):
+            on, off = self._create_and_filter_onofflists(t_index=t_index, energy_range=energy_range)
+
+            for time in on.time:
+                time_holder.append([time.tt.mjd, 'on'])
+            for time in off.time:
+                time_holder.append([time.tt.mjd, 'off'])
+
+        # sort all elements in the table by time
+        time_holder = sorted(time_holder, key=lambda item: item[0])
+        time_holder = np.asarray(time_holder)
+        # print(time_holder)
+
+        rows = []
+        istart = 1
+        i = 1
+        n = 0
+        while time_holder[i][0] < time_holder[-1][0]:
+            i += 1
+            if time_holder[i][1] == 'break':
+                while time_holder[i + 1][1] != 'on' and time_holder[i + 1][1] != 'off':
+                    i += 1
+                n += np.sum(time_holder[istart:i] == 'end')
+                istart = i
+                continue
+            if time_holder[i][1] != 'on' and time_holder[i][1] != 'off':
+                continue
+
+            # compute alpha
+            alpha = self._alpha(time_holder, obs_properties, n, istart, i)
+
+            non = np.sum(time_holder[istart:i + 1] == 'on')
+            noff = np.sum(time_holder[istart:i + 1] == 'off')
+
+            # check the significance
+            signif = significance_on_off(non, noff, alpha, method=significance_method)
+            if signif > significance:
+                rows.append(dict(
+                    t_start=Time((float(time_holder[istart - 1][0]) + float(time_holder[istart][0])) / 2, format="mjd",
```

Comment:
I guess it's not a real efficiency concern here, but still, I'd suggest you just store the floats in the dict here, and then after you've made your table, convert back to Time objects with one call processing arrays.
```
table = Table(rows=rows)
table['t_start'] = Time(table['t_start'], format='mjd', scale='tt')
table['t_stop'] = Time(table['t_stop'], format='mjd', scale='tt')
```

Reply:
Is it useful? Because if it change nothing it would only add 3 lines of code

Reply:
See my comment above. I do think structuring the code to do Time <-> float only at the start and end would be a bit simpler, and faster than having conversions in the middle of the algorithm. But I agree it's not a huge win, more a small preference. So you decide.